src/OrleansRuntime/Streams/PubSub/PubSubRendezvousGrain.cs

```diff
@@ -281,7 +281,8 @@ private void RemoveProducer(PubSubPublisherState producer)
         {
             counterConsumersRemoved.Increment();
             if (State.Consumers.Any(c => c.IsFaulted && c.Equals(subscriptionId)))
-                throw new FaultedSubscriptionException(subscriptionId, streamId);
+                logger.Warn(ErrorCode.Stream_RemoveFaultedSubscription,
```

Comment:
Originally faulted subscriptions were intended to be permanent, but it should probably be ok to remove them via the subscription management interfaces or by unsubscribing.  That being the case, I don't think we even need to check (and log) in this case.  So we can just remove this entire if block.